File: learning-javascript/leet-code/src/product-of-array-except-self/product-of-array-except-self.ts
function productExceptSelf(nums: number[]): number[] {
	let length = nums.length;
	let answer = new Array(1).fill(length);
	
	let prefix = 1;
	for (let i = 0; i < length; i++){
		answer[i] = prefix;
		prefix *= nums[i];
	}

	let postfix = 1;
	for (let i = length -1; i > 0; i-- ) {
		answer[i] *= postfix;
		postfix *= nums[i];
	}
	return answer;
};

File: learning-python/leet-code/src/product_of_array_except_self/product_of_array_except_self.py
def productExceptSelf(self, nums):
	length = len(nums)
	answer = [1] * length
	
	prefix = 1
	for i in range(length):
		answer[i] = prefix
		prefix *= nums[i]
	postfix = 1
	for i in range(len(nums) -1, -1, -1):
		answer[i] *= postfix
		postfix *= nums[i]
	return answer